features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/DefaultRecorderProvider.kt

```diff
@@ -98,7 +100,7 @@ internal class DefaultRecorderProvider(
             MapperTypeWrapper(
                 SwitchCompat::class.java,
                 SwitchCompatMapper(
-                    textViewMapper,
+                    textViewMapper as TextViewMapper<SwitchCompat>,
```

Comment:
this part with explicit casting looks dangerous and not very obvious (partially because it is not obvious that `SwitchCompat`, etc. are sub-classes of `TextView`). Is there a way to avoid explicit cast?

Reply:
Those cast are totally safe, but I agree they are not obvious. 
They are temporary though, the next PRs will be about letting `SwitchCompatMapper`(and others) extend `TextViewMapper` instead.